runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -136,8 +129,7 @@ public final synchronized ILogData nextUpTo(final long maxGlobal) {
         }
 
         // Get the next entry from the underlying implementation.
-        final ILogData entry =
-                getNextEntry(getCurrentContext(), maxGlobal);
+        ILogData entry = streamReadWithCheck(() -> getNextEntry(getCurrentContext(), maxGlobal), maxGlobal);
```

Comment:
what happens if we are rolling back, i.e., we have already resolved in the past for the "compacted addresses" and it's is in our cache (undo records that take us to that snapshot).. do we still abort (throw trimmedExceptions?) regardless of this?

Reply:
Yes, an TrimmedException is thrown for the rolling back case. There are some reason for this design decision. 

1. Addresses before compaction mark may have been trim to prevent memory leak. In this case, rolling back is limit by compaction mark to insure correctness.

2. Let us assume we do not trim resolved addresses in the stream layer. Aggressively throwing trimmedException is still beneficial to some extent. Rolling back is implemented by applying undo record one by one. In the middle of rolling back, if the client needs to read data from the log unit if the data is not presented in the case, the client will throw trimmedException which make the previous effort in vain.